app/controllers/users/sessions_controller.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module Users
+  class SessionsController < Devise::SessionsController
+    def new
+      if params.key?(:email)
+        self.resource = User.new(email: params[:email])
+      else
+        super
+      end
+    end
+
+    protected
```

Comment:
```suggestion
    private
```

Learn about `protected` in Ruby. It's not the same as in the common OOP (like Java).

Reply:
I'm aware about. It's just missprint